src/components/Reactions/Reaction.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+
+import {Button, ButtonSize, PaletteOption, PopoverProps, Popup} from '@gravity-ui/uikit';
+
+import {block} from '../utils/cn';
+import {useStableCallback} from '../utils/useStableCallback';
+
+import {useReactionsContext} from './context';
+import {useReactionsPopup} from './hooks';
+
+export interface ReactionProps extends PaletteOption {
+    /**
+     * Should be true when the user used this reaction.
+     */
+    selected?: boolean;
+    /**
+     * Display a number after the icon.
+     * Represents the number of users who used this reaction.
+     */
+    counter?: React.ReactNode;
+    /**
+     * If present, when a user hovers over the reaction, a popover appears with `tooltip.content`.
+     * Can be used to display users who used this reaction.
+     */
+    tooltip?: ReactionTooltipProps;
+}
+
+export interface ReactionTooltipProps
+    extends Pick<PopoverProps, 'strategy' | 'placement' | 'modifiers'> {
+    /**
+     * Tooltip's content.
+     */
+    content: React.ReactNode;
+    /**
+     * Tooltip content's HTML class attribute.
+     */
+    className?: string;
+    /**
+     * Fires when the `onMouseLeave` callback is called.
+     * Usage example:
+     * you have some popup inside a tooltip, you hover on it, you don't want the tooltip to be closed because of that.
+     */
+    canClosePopup?: () => boolean;
+}
+
+interface ReactionInnerProps {
+    reaction: ReactionProps;
+    size: ButtonSize;
+    onClick?: (value: string) => void;
+}
+
+const popupDefaultPlacement: PopoverProps['placement'] = [
+    'bottom-start',
+    'bottom',
+    'bottom-end',
+    'top-start',
+    'top',
+    'top-end',
+];
+
+const b = block('reactions');
+
+export function Reaction(props: ReactionInnerProps) {
+    const {value, disabled, selected, content, counter, tooltip} = props.reaction;
+    const {size, onClick} = props;
+
+    const onClickCallback = useStableCallback(() => onClick?.(value));
```

Comment:
Why it's not just `useCallback`?

Reply:
I dont get why in this particular case you dont want to use useCallback

Reply:
`useStableCallback` seems to be a better `useCallback`
But I don't insist on using it here
I'll return the useCallback if you want to

Reply:
Removed `useStableCallback`